src/bridge/views/InviteTicket.js

```diff
@@ -144,8 +146,13 @@ class InviteTicket extends React.Component {
       let handler = isLogin ? this.handleInviteTicketInput : this.handleVerifyTicketInput
       let label = isLogin ? 'Activation code' : 'ticket'
 
+      const TicketElem =
```

Comment:
Could consider this, which unconditionally avoids the naked `.value` access:

```
// if realWallet is Just, then VerifyTicketInput appropriate contents; if Nothing, remain Nothing
const Verified = this.state.realWallet.map(realWallet =>
  VerifyTicketInput(realWallet.ticket)
)

// Verified.getOrElse(TicketInput) falls back to TicketInput if Verified is Nothing
const TicketElem =
    isVerify
  ? Verified.getOrElse(TicketInput)
  : TicketInput
```

Reply:
Oh, very cool. Hadn't seen the `map` function before.